docs/tapable-hooks.md

```diff
@@ -52,103 +39,119 @@ sendHTTPResponse        // 发送 HTTP 响应
 
 ### hooks.render
-渲染服务端 App 时调用，对于 Vue 应用，此步骤对应加载 vue-render-bundle 渲染页面。这个钩子主要是给  server  实现方使用，并行执行，第一个返回的结果即为渲染结果。渲染结果 RenderResult 支持多种类型，包括渲染失败等情况。除非特殊情况 app 业务实现方应该忽略此 hook。
+HTTP server should tap this hook, import app's render bundle and render the HTML with data in `RenderContext`. Applications generally don't use this hook unless custom rendering is expected.
 
-| 类别 | Description |
+| Key | Content |
 | ----------- | ----------- |
-| 名称 | render |
-| Hook 类型 | AsyncParallelBailHook |
-| 参数类型 | RenderContext |
-| 返回值类型 | RenderResult |
+| Name | render |
+| Hook Type | AsyncParallelBailHook |
+| Parameters | RenderContext |
+| Return Value | RenderResult |
 
 ### hooks.afterRender
-在 App 渲染完成后执行，根据渲染成功或失败，RenderResult 可能为成功或失败，如需重载渲染结果，钩子可以返回更改后的 RenderResult。渲染结果中包含 SsrBundleResult，服务端会使用 SsrBundleResult 中的值重新拼装页面模板。这里可以简单替换掉页面 template 而不引起 hydration mismatch (模板是 Vue app 以外的部分)。注意钩子瀑布流顺序执行。
+Tapped functions will be called one by one after render finishes, processing the result of a successful rendering or handle the error of a failed rendering.
+Failed SSR could be downgraded to CSR here.
+Be careful of [hydration mismatch].
 
-| 类别 | Description |
+| Key | Content |
 | ----------- | ----------- |
-| 名称 | afterRender |
-| Hook 类型 | AsyncSeriesWaterfallHook |
-| 参数类型 | RenderResult |
-| 返回值类型 | RenderResult |
+| Name | afterRender |
+| Hook Type | AsyncSeriesWaterfallHook |
+| Parameters | RenderResult |
+| Retuan Value | RenderResult |
 
 ### hooks.beforeResponse
-在所有 HTTP 响应发送前执行，任意回调函数(tapped function)返回结果即终止其它回调函数执行。优先返回 HTTPResponse 将替代原有 HTTPResponse 返回。参数 RenderResult 包含 RenderContext 中各钩子添加的 meta data 和渲染异常 Error 等信息，可通过它们构建最终响应 HTTPResponse。
+Tapped functions will be called after `afterRender`. This is the last hook before HTTP Response is sent and it's mainly used for compose the HTTP Response, which could be generated with all data in the `RenderResult` which contains the `RenderContext`.
+
+Any tapped function returns an HTTPResponse will invalidate the return value of other tapped functions.
 
-| 类别 | Description |
+| Key | Content |
 | ----------- | ----------- |
-| 名称 | beforeResponse |
-| Hook 类型 | AsyncParallelBailHook |
-| 参数类型 | RenderResult |
-| 返回值类型 | HTTPResponse \| void |
-
-## 使用示例
-### App 开发者
-App 开发者首先使用 [Vise 命令行工具](./commandline-tool.html)新建自己的业务项目（即 App），得到如 [App目录结构](./start-develop.html#app-目录结构) 中描述的默认项目文件夹。
-在项目的 `app-my-project/src/server-hooks.ts` 文件中，开发者可以在从服务器接收到用户 HTTP 请求到发出 HTTP 响应的 9 个 hooks 生命周期中添加自己的业务逻辑。
-#### server-hooks.ts 配置文件
-一份典型的 `server-hooks.ts` 文件内容如下：
+| Name | beforeResponse |
+| Hook Type | AsyncParallelBailHook |
+| Parameters | RenderResult |
+| Return Value | HTTPResponse \| void |
+
+## Hooks Usage
+### App developer
+App developer use [Vise command line tool](./commandline-tool.html) to created scaffold of an app first, then there will be a app directory described in [App Directory Structure](./start-develop.html#app-directory-structure).
```

Comment:
App developer can use [Vise command line tool](./commandline-tool.html) to create the scaffold of an app, then there will be an app directory as described in [App Directory Structure](./start-develop.html#app-directory-structure).

Reply:
Terrific, fixed.